src/test/java/Homework22.java

```diff
@@ -0,0 +1,20 @@
+import Pages.HomePage;
+import Pages.LoginPage;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class Homework22 extends BaseTest{
+
+    @Test
+    public void playSongTest(){
+        LoginPage loginpage = new LoginPage(driver);
+        HomePage homepage = new HomePage(driver);
+        loginpage.login();
+
+        homepage.getPlayNextSong();
+        homepage.getPlay();
```

Comment:
You got this problem, please correct it. Please finish your homework.

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/a0a613f0-ab59-4090-9b2e-3a06c516f47b)

HINT

**Use Class Actions** https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/interactions/Actions.html

![18](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/6a56dc23-e410-4afd-9e86-80f5689a83dc)

**or**

if your problem is still not solved, you need a DOM-level solution. So, we need an interface that helps to execute JavaScript with Selenium WebDriver. It is called JavaScriptExecutor. JavaScript works at the DOM level. Therefore, we are able to click any web element by using JavaScript executor.

```
 @When("I push the button play")
    public void iPushTheButtonPlay() throws InterruptedException {

        String playNextButtonLocator = "//i[@data-testid='play-next-btn']";
        WebElement playNextButtonElement = driver.findElement(By.xpath(playNextButtonLocator));

        JavascriptExecutor jsExecutor = (JavascriptExecutor) driver;
        jsExecutor.executeScript("arguments[0].scrollIntoView(true);", playNextButtonElement);
        jsExecutor.executeScript("arguments[0].click();", playNextButtonElement);

        String playButtonLocator = "//span[@data-testid='play-btn']";
        WebElement playButtonElement = driver.findElement(By.xpath(playButtonLocator));
        jsExecutor.executeScript("arguments[0].scrollIntoView(true);", playButtonElement);
        jsExecutor.executeScript("arguments[0].click();", playButtonElement);
    }
```

![image](https://github.com/timurk8/A47-QA-Automation-Framework/assets/104761482/7c710d85-ab3c-4533-bedb-8d8dd9cb5bd5)


**Clicking Elements in Selenium using JavaScript** (https://www.baeldung.com/java-selenium-javascript)

And I added  `wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("img[alt='Avatar of Timur']")));`

![image](https://github.com/timurk8/A47-QA-Automation-Framework/assets/104761482/4311515d-2af1-4d81-9fe2-1c78e9f64789)

![18_1](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/4abfe7a0-a027-470e-aa17-140786fc626c)